tests/sqlparser_common.rs

```diff
@@ -12577,9 +12577,20 @@ fn overflow() {
     let statement = statements.pop().unwrap();
     assert_eq!(statement.to_string(), sql);
 }
-
 #[test]
 fn parse_select_without_projection() {
     let dialects = all_dialects_where(|d| d.supports_empty_projections());
     dialects.verified_stmt("SELECT FROM users");
 }
+
+#[test]
+fn parse_update_from_before_select() {
+    all_dialects()
+    .verified_stmt("UPDATE t1 FROM (SELECT name, id FROM t1 GROUP BY id) AS t2 SET name = t2.name WHERE t1.id = t2.id");
+
+    let res = all_dialects().parse_sql_statements(
+        "UPDATE t1 FROM (SELECT name, id FROM t1 GROUP BY id) AS t2 SET name = t2.name FROM (SELECT name from t2) AS t2",
+    );
+
+    assert!(res.is_err());
```

Comment:
could we assert in [this style](https://github.com/apache/datafusion-sqlparser-rs/blob/6cdf66457006a3fbfc569dc262e78f404522821b/tests/sqlparser_common.rs#L219-L235) to expect a specific error message?